src/SourceLocator/SourceStubber/ReflectionSourceStubber.php

```diff
@@ -76,21 +76,20 @@ public function __construct()
      */
     public function generateClassStub(string $className): ?StubData
     {
-        /** phpcs:disable SlevomatCodingStandard.Namespaces.ReferenceUsedNamesOnly.ReferenceViaFullyQualifiedName */
-        $isEnum = function_exists('enum_exists') && \enum_exists($className, false);
-        /** phpcs:enable */
-
         if (
             ! (
             class_exists($className, false)
             || interface_exists($className, false)
             || trait_exists($className, false)
-            || $isEnum
             )
         ) {
             return null;
         }
 
+        /** phpcs:disable SlevomatCodingStandard.Namespaces.ReferenceUsedNamesOnly.ReferenceViaFullyQualifiedName */
+        $isEnum = function_exists('enum_exists') && \enum_exists($className, false);
```

Comment:
do enums pass `class_exists()` checks?

Reply:
Yes. I just figured it out. 🤷‍♂️ 

Here I need preserve the check to choose `ReflectionEnum` or `ReflectionClass`.